rclcpp_examples/src/parameters/ros2param.cpp

```diff
@@ -125,6 +136,21 @@ int main(int argc, char ** argv)
       fprintf(stderr, "Failed to set parameter\n");
       return 1;
     }
+  } else if (op == PARAM_LIST) {
+    auto list_parameters_result = parameters_client->list_parameters({},10);
+    auto list_result = rclcpp::spin_until_future_complete(
+        node, list_parameters_result, std::chrono::milliseconds(10000));
+    if (list_result == rclcpp::executor::FutureReturnCode::SUCCESS) {
+      printf("Node %s has %d parameters:\n", remote_node.c_str(), (int) list_parameters_result.get().names.size());
+      for (auto name : list_parameters_result.get().names) {
+        printf("%s\n", name.c_str());
+      }
+    } else if (list_result == rclcpp::executor::FutureReturnCode::TIMEOUT){
+      printf("Timed out trying to list parameters from node\n");
```

Comment:
It would be good if the timeout were configurable, but at the very lest it should print out how long it waited in the timeout exception message.